src/game/shell.c

```diff
@@ -194,20 +275,40 @@ bool Shell_MakeScreenshot()
         break;
     }
 
-    char path[20];
-    for (int i = 0; i < 10000; i++) {
-        sprintf(path, "screenshot%04d.%s", i, ext);
+    // Screenshot folder
+    char ss_folder[SS_FOLDER_NAME_SIZE];
+    sprintf(ss_folder, "screenshots");
+    File_CreateDirectory(ss_folder);
+
+    // Screenshot name
+    char ss_name[TIMESTAMP_SIZE + LEVEL_TITLE_SIZE];
+    Shell_GetScreenshotName(ss_name);
+
+    // Screenshot folder/name path
+    char path
+        [SS_FOLDER_NAME_SIZE + TIMESTAMP_SIZE + LEVEL_TITLE_SIZE
+         + EXTENSION_SIZE];
+    sprintf(path, "%s/%s.%s", ss_folder, ss_name, ext);
+    char *full_path = NULL;
+    File_GetFullPath(path, &full_path);
+    if (!File_Exists(full_path)) {
+        bool result = Output_MakeScreenshot(full_path);
+        Memory_FreePointer(&full_path);
+        return result;
+    }
+
+    // Name already exists so add number to name
+    for (int i = 2; i < 1000; i++) {
+        sprintf(path, "%s/%s_%d.%s", ss_folder, ss_name, i, ext);
         char *full_path = NULL;
         File_GetFullPath(path, &full_path);
-
         if (!File_Exists(full_path)) {
             bool result = Output_MakeScreenshot(full_path);
             Memory_FreePointer(&full_path);
             return result;
         }
-
-        Memory_FreePointer(&full_path);
     }
 
+    Memory_FreePointer(&full_path);
```

Comment:
This is now in the wrong place, we allocate memory with `File_GetFullPath` in each iteration but free it only once after all iterations have finished.